db/migrate/20160403055927_create_deployment_authentications.rb

```diff
@@ -0,0 +1,33 @@
+class CreateDeploymentAuthentications < ActiveRecord::Migration[5.0]
+  def change
+    create_table :deployment_authentications do |t|
+      t.string :name
+      t.boolean :challenge
+      t.boolean :login
+      t.string :kind
+      t.string :htpassd_file_path
+      t.text :htpassd_users
+      t.text :ldap_id, :array => true, :default => []
+      t.text :ldap_email, :array => true, :default => []
+      t.text :ldap_name, :array => true, :default => []
+      t.text :ldap_preferred_user_name, array: true, :default => []
+      t.string :ldap_bind_dn
+      t.string :ldap_bind_password
+      t.string :ldap_ca
+      t.boolean :ldap_insecure
+      t.string :ldap_url
+      t.string :request_header_challenge_url
+      t.string :request_header_login_url
+      t.string :request_header_client_ca
+      t.text :request_header_headers, :array => true, :default => []
```

Comment:
also the request header `preferredUsernameHeaders`, `nameHeaders`, and `emailHeaders`?